tests/publiccloud/create_aistack_env.pm

```diff
@@ -181,13 +232,24 @@ sub test_openwebui_service {
     set_var('OPENWEBUI_IP', "$ipaddr");
     record_info("Added $ipaddr to /etc/hosts with hostname $host_name");
 
-    # connect open-webui service
-    my $curl_cmd = "curl -v -k https://$host_name";
-    my $curl_result = script_run($curl_cmd);
-    if ($curl_result == 0) {
-        record_info("Successfully connected to the open-webui service at $curl_cmd \n");
+    # get endpoints
+    assert_script_run("kubectl get endpoints $sr_name -n $namespace -o=jsonpath='{.subsets[*].addresses[*].ip}'");
+    my $endpoint_cmd = "kubectl get endpoints $sr_name -n $namespace -o=jsonpath='{.subsets[*].addresses[*].ip}'";
+    my $endpoint_result = script_output($endpoint_cmd);
+    record_info("Endpoint code: $endpoint_result \n");
+    if (!$endpoint_result) {
+        die "No healthy endpoints found for the open-webui service in $namespace\n";
     } else {
-        die "Unable to connect to the open-webui service at $curl_cmd\n";
+        # connect open-webui service
+        assert_script_run("curl --output /dev/null --silent --head --write-out \"%{http_code}\n\" -k -L https://$host_name");
+        my $curl_cmd = "curl --output /dev/null --silent --head --write-out \"%{http_code}\n\" -k -L https://$host_name";
+        my $curl_result = script_output($curl_cmd);
+        record_info("http code: $curl_result \n");
+        if ($curl_result == 200) {
+            record_info("Successfully connected to the open-webui service at $curl_cmd \n");
+        } else {
+            die "Received unexpected HTTP error code $curl_result for $curl_cmd\n";
+        }
```

Comment:
Hi Yoga, please, in last commit you also added the `assert_...`, but that is the same command done below, then **repeated twice**, even _not needed_, because you already do the check of output, after script_output executed.

In a _next Merge request_, please, consider this kind update:

remove it:
~~assert_script_run("curl --output /dev/null --silent --head --write-out \"%{http_code}\n\" -k -L https://$host_name");~~

and change the full block `L244-L252` like that:
```
record_info("http check”, $host_name);
my $curl_cmd = "curl --output /dev/null --silent --head --write-out \"%{http_code}\n\" -k -L https://$host_name";
validate_script_output("$curl_cmd", qr/200$/, fail_message => “Received unexpected HTTP error code from: $curl_cmd”);
```
where the `$` in `/200$/` expects that value aligned at end of the output.

Thank you.